### PR TITLE
Add Navigation Abstractions APIs

### PR DESCRIPTION
**Pull request by hishamco**
Move some of navigation types into navigation abstractions project

### CODE REVIEW

File: src/OrchardCore/OrchardCore.Navigation.Core/OrchardCore.Navigation.Core.csproj [hunk 1]
```diff
@@ -14,7 +14,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <ProjectReference Include="..\OrchardCore.Infrastructure.Abstractions\OrchardCore.Infrastructure.Abstractions.csproj" />
```
[the remark this answers is not in the document]
Comment (hishamco) [reply src/OrchardCore/OrchardCore.Navigation.Core/OrchardCore.Navigation.Core.csproj:17]: Exactly, I just fix it and move Pager related types into abstractions, the commit is ready in my local. I will push it soon

**Comment by deanmarcussen** [comment]
How many files are actually left in the Core project after this?

**Comment by hishamco** [comment]
The implementation types, which are 2 or 3 I think, this happen after we moved pager related types. If anyone disgree with move pager types I will move them back

**Comment by hishamco** [comment]
I don't think it's a matter of no. of file. OC.Email.Core for instance has only one services, but the separation make them clear. So my doubt now about pager related types coz OC.List module are related to them

**Review by Piedone — CHANGES_REQUESTED** [review]
This would break like a third of the modules out there, since you'd need to add a new package reference to every module having a menu.

We shouldn't move around things just for the sake of it without much consideration. I've seen you do a lot of such stuff Hisham, and I'd discourage you from doing this: refactoring to remove clutter is good, but only if you involve the feedback of others (i.e. is it an issue that others agree with being an issue?), and it's non-breaking for our users or the refactoring is so much in-depth and useful that it makes it worthwhile. Moving around a few classes to do some polishing is not worth it. I can empathize with the sentiment, as I want things to be tidy too, but with a project like OC we need to consider the bigger picture.

**Comment by Piedone** [comment]
Breaking changes are sometimes necessary, but we should only have them when they are indeed necessary and enable something important. Not for small clean-ups.

**Comment by Piedone** [comment]
BTW I suggest you open issues instead of PRs for such things first, because then we can discuss it before you put time into doing them.

And this PR I suggest closing.

**Pull request closed without merging**